lib/instance.js

```diff
@@ -754,13 +754,23 @@ module.exports = (function() {
    * @alias updateAttributes
    */
   Instance.prototype.update = function(values, options) {
+    var changeMemory = this.changed() || []
+      , affectedBySetter
+      , valuesShouldUpdate;
+
     options = options || {};
     if (Array.isArray(options)) options = {fields: options};
 
     this.set(values, {attributes: options.fields});
 
+    // Now we need to figure out which fields were actually affected by the setter.
+    // We build a somewhat weird argument list array that we can apply to _.without
+    changeMemory.unshift(this.changed());
+    affectedBySetter = _.without.apply(this, changeMemory);
```

Comment:
Can be expressed more cleanly with `[this.changed()].concat(changeMemory)` i think.